agent/host_agent_test.go

```diff
@@ -190,7 +233,88 @@ var _ = Describe("Agent", func() {
 			byoHost := builder.ByoHost(ns.Name, "random-second-host").Build()
 			Expect(k8sClient.Create(context.TODO(), byoHost)).NotTo(HaveOccurred(), "failed to create byohost")
 
-			Consistently(session.Err, "10s").ShouldNot(gbytes.Say(byoHost.Name))
+			defer output.Close()
+
+			f := e2e.WriteDockerLog(output, agentLogFile)
+			defer func() {
+				deferredErr := f.Close()
+				if deferredErr != nil {
+					e2e.Showf("error closing file %s: %v", agentLogFile, deferredErr)
+				}
+			}()
+			Consistently(func() (done bool) {
+				_, err := os.Stat(agentLogFile)
+				if err == nil {
+					data, err := os.ReadFile(agentLogFile)
+					if err == nil && strings.Contains(string(data), byoHost.Name) {
+						return true
+					}
+				}
+				return false
+			}, 10, 1).ShouldNot(BeTrue())
+		})
+		Context("when machineref & bootstrap secret is assigned", func() {
+			var (
+				byoMachine *infrastructurev1beta1.ByoMachine
+				namespace  types.NamespacedName
+			)
+			BeforeEach(func() {
+				byoMachine = builder.ByoMachine(ns.Name, defaultByoMachineName).Build()
+				Expect(k8sClient.Create(ctx, byoMachine)).Should(Succeed())
+				byoHost := &infrastructurev1beta1.ByoHost{}
+				namespace = types.NamespacedName{Name: hostName, Namespace: ns.Name}
+				Eventually(func() (err error) {
```

Comment:
Can we replace this `Eventually` with this function call?
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/6b69e60f40f964c03f429e0acf3e810804d5e268/controllers/infrastructure/suite_test.go#L146

Reply:
Although we might want to revisit writing / reading to cache. I don't think we read from cache anymore.
cc @dharmjit

Reply:
I think this requires a handle to the `reconciler`, which we do not have here.
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/6b69e60f40f964c03f429e0acf3e810804d5e268/controllers/infrastructure/suite_test.go#L151